lib/server.go

```diff
@@ -1400,6 +1415,27 @@ func (srv *Server) _handleSnapshot(pp *Peer, msg *MsgDeSoSnapshotData) {
 	srv.GetBlocks(pp, int(headerTip.Height))
 }
 
+// updateDbOpts closes the current badger DB instance and re-opens it with the provided new options.
+func (srv *Server) updateDbOpts(opts badger.Options) {
+	// Make sure that a mempool process doesn't try to access the DB while we're closing and re-opening it.
+	srv.mempool.mtx.RLock()
+	defer srv.mempool.mtx.RUnlock()
+	// Make sure that a server process doesn't try to access the DB while we're closing and re-opening it.
+	srv.DbMutex.Lock()
+	defer srv.DbMutex.Unlock()
+	srv.blockchain.db.Close()
+	db, err := badger.Open(opts)
+	if err != nil {
+		// If we can't open the DB with the new options, we need to exit the process.
+		glog.Fatalf("Server._handleSnapshot: Problem switching badger db to performance opts, error: (%v)", err)
+	}
+	srv.blockchain.db = db
+	srv.snapshot.mainDb = srv.blockchain.db
+	srv.mempool.bc.db = srv.blockchain.db
+	srv.mempool.backupUniversalUtxoView.Handle = srv.blockchain.db
+	srv.mempool.universalUtxoView.Handle = srv.blockchain.db
```

Comment:
This is dirty but I'm trying to think about how we would clean it up. We could have a wrapper on the DB that this thing can update... but that's dirty too. Ideally you could change the db without having to update all of these handles manually... I'm noodling.

Reply:
Agreed, not exactly a pretty solution, but the only other option I could think of was the wrapper struct, which you've mentioned.